packages/core/src/electron-browser/window/electron-window-preferences.ts

```diff
@@ -0,0 +1,62 @@
+/********************************************************************************
+ * Copyright (C) 2021 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { interfaces } from 'inversify';
+import { createPreferenceProxy, PreferenceContribution, PreferenceProxy, PreferenceSchema, PreferenceService } from '../../browser/preferences';
+
+export namespace ZoomLevel {
+    export const DEFAULT = 0;
+    // copied from https://github.com/microsoft/vscode/blob/dda96b69bfc63f309e60cfc5f98cb863c46b32ac/src/vs/workbench/electron-sandbox/actions/windowActions.ts#L47-L48
+    export const MIN = -8;
+    export const MAX = 9;
+    // amount to increment or decrement the window zoom level.
+    export const VARIATION = 0.5;
+}
+
+export const electronWindowPreferencesSchema: PreferenceSchema = {
+    type: 'object',
+    properties: {
+        'window.zoomLevel': {
+            'type': 'number',
+            'default': ZoomLevel.DEFAULT,
+            'minimum': ZoomLevel.MIN,
+            'maximum': ZoomLevel.MAX,
+            'scope': 'application',
+            // eslint-disable-next-line max-len
+            'description': 'Adjust the zoom level of the window. The original size is 0 and each increment above (e.g. 0.5) or below (e.g. -0.5) represents zooming 10% larger or smaller. You can also enter decimals to adjust the zoom level with a finer granularity.'
```

Comment:
I'm intrigued as to why you changed the description to say 0.5 is 10%, when both vs-code and the Electron documentation say a level of 1 changes the zoom by a factor of 1.2.  They are not the same (1.1 x 1.1 != 1.2), and surely it is clearer so say what a level of one does?

Technically I believe the vs-code description is incorrect.  Increasing the level by one adds 20%, decreasing by one deducts 25%.